modular_azurepeak/code/modules/spells/pantheon/inhumen/baotha.dm

```diff
@@ -132,3 +132,224 @@
+	icon_state = "grabbing_greyscale"
+	color = "#FFFFFF"
+	var/right_click = FALSE
+
+/obj/item/melee/touch_attack/loversruin/attack_self()
+	qdel(src)
+
+/obj/item/melee/touch_attack/loversruin/afterattack(atom/target, mob/living/carbon/human/user, proximity)
+	if(istype(user.used_intent, /datum/intent/fill) && create_ichor(target, user))
+		qdel(src)
+
+/datum/reagent/medicine/loversruin
+	name = "Lover's Ruin"
+	description = "A sweet smelling concoction. It has small charred petals swimming on the surface."
+	color = "#9c2745"
+	taste_description = "sin"
+
+/datum/reagent/medicine/loversruin/on_mob_life(mob/living/carbon/M)
+	if(HAS_TRAIT(M, TRAIT_CRACKHEAD))
+		if(volume >= 60)
+			M.reagents.remove_reagent(/datum/reagent/medicine/loversruin, 2)
+		if(M.blood_volume < BLOOD_VOLUME_NORMAL)
+			M.blood_volume = min(M.blood_volume+40, BLOOD_VOLUME_MAXIMUM)
+		var/list/wCount = M.get_wounds()
+		if(wCount.len > 0)
+			M.heal_wounds(4.5)
+		if(volume > 0.99)
+			M.adjustBruteLoss(-2*REM, 0)
+			M.adjustFireLoss(-2*REM, 0)
+			M.adjustOxyLoss(-2, 0)
+			M.adjustToxLoss(-2, 0)
+			M.adjustOrganLoss(ORGAN_SLOT_BRAIN, -5*REM)
+			M.adjustCloneLoss(-4*REM, 0)
+	else
+		M.adjustToxLoss(3, 0)
+		M.adjustOxyLoss(1, 0)
+	..()
+
+/obj/item/melee/touch_attack/loversruin/proc/create_ichor(atom/thing, mob/living/carbon/human/user)
+	if(!thing.Adjacent(user))
+		to_chat(user, span_info("I need to be closer to [thing] in order to try filling it with the concoction."))
+		return
+
+	if(thing.is_refillable())
+		if(thing.reagents.holder_full())
+			to_chat(user, span_warning("[thing] is full."))
+			return
+
+		user.visible_message(span_info("[user] closes [user.p_their()] eyes in prayer and extends a hand over [thing] as a sweet smelling ichor begins to stream from [user.p_their()] fingertips..."), span_notice("I call forth [user.patron.name], to fill [thing] with Her blessings..."))
+
+		var/holy_skill = user.get_skill_level(attached_spell.associated_skill)
+		var/drip_speed = 56 - (holy_skill * 8)
+		var/fatigue_spent = 0
+		var/fatigue_used = max(3, holy_skill)
+		while(do_after(user, drip_speed, target = thing))
+			if(thing.reagents.holder_full() || (user.devotion.devotion - fatigue_used <= 0))
+				break
+
+			var/water_qty = max(1, holy_skill) + 1
+			var/list/water_contents = list(/datum/reagent/medicine/loversruin = water_qty)
+			var/datum/reagents/reagents_to_add = new()
+			reagents_to_add.add_reagent_list(water_contents)
+			reagents_to_add.trans_to(thing, reagents_to_add.total_volume, transfered_by = user, method = INGEST)
+
+			fatigue_spent += fatigue_used
+			user.stamina_add(fatigue_used)
+			user.devotion?.update_devotion(-1.0)
+
+			if(prob(80))
+				playsound(user, 'sound/items/fillcup.ogg', 55, TRUE)
+
+		return max(50, fatigue_spent)
+	else
+		to_chat(user, span_info("I'll need to find a container that can hold Her blessing."))
+
+//T1, Baotha's version of Eora's Bud (now renamed True Peace Bloom). Applies the TRAIT_CRACKHEAD baothans have.
```

Comment:
should we be giving out TRAIT_CRACKHEAD and +1 END/CON like this? 10 baothans can pass this to 20 people in an hour if they're diligent and the game is simply not ready to have that many people running around tweaking out of their minds

Reply:
no longer gives stat bonus, the crackhead trait is good enough (as this lets baothans mark their allies, and the crackhead trait has a few advantages of its own like being able to drink the baothan juice to heal)